(maint) Move helpers to utils namespace

justinstoller opened the pull request:
Previously we included middleware and utility methods all in the core
namespace.

This was awkward for consumers as they often only need one or the other.

This duplicates out all of the public utility functions into a utils
namespace, sets the core functions to depend on the them, and updates
the test accordingly. It leaves the existing implementations in the core
namespace so that consumers can upgrade components without having to
move in lockstep.

KevinCorcoran commented:
>  It leaves the existing implementations in the core namespace so that consumers can upgrade components without having to move in lockstep.

I think it would be better to just move them instead of copy.  Having two implementation is technical debt.  I don't think there are enough consumers of these functions for it to be a big deal; this library is not yet 1.0 anyway.

justinstoller commented:
Updated to move without duplicating the helper functions.
Also rebased onto https://github.com/puppetlabs/ring-middleware/pull/34 so that that PR can move ahead of this one.

KevinCorcoran commented:
👍

justinstoller commented:
@nathanielksmith I believe, per @KevinCorcoran 's comment above, that we're fine getting this into the next release of r-m. If you could review, I think you can merge if satisfied.
